TombLib/TombLib/LevelData/Compilers/TombEngine/TombEngineSpriteAllocator.cs

```diff
@@ -0,0 +1,290 @@
+
+            for (int xP = 0; xP < padding; xP++)
+            {
+                // copy left line
+                if (xP < padding)
+                    to.CopyFrom(position.X + xP, dataOffset + position.Y + padding, from,
+                               x, y, 1, height);
+
+                // copy right line
+                if (xP < padding)
+                    to.CopyFrom(position.X + xP + width + padding, dataOffset + position.Y + padding, from,
+                               x + width - 1, y, 1, height);
+
+                for (int yP = 0; yP < padding; yP++)
+                {
+                    // copy top line
+                    if (yP < padding)
+                        to.CopyFrom(position.X + padding, dataOffset + position.Y + yP, from,
+                                   x, y, width, 1);
+                    // copy bottom line
+                    if (yP < padding)
+                        to.CopyFrom(position.X + padding, dataOffset + position.Y + yP + height + padding, from,
+                                   x, y + height - 1, width, 1);
+
+                    // expand top-left pixel
+                    if (xP < padding && yP < padding)
+                        to.SetPixel(position.X + xP, dataOffset + position.Y + yP, topLeft);
+                    // expand top-right pixel
+                    if (xP < padding && yP < padding)
+                        to.SetPixel(position.X + xP + width + padding, dataOffset + position.Y + yP, topRight);
+                    // expand bottom-left pixel
+                    if (xP < padding && yP < padding)
+                        to.SetPixel(position.X + xP, dataOffset + position.Y + yP + height + padding, bottomLeft);
+                    // expand bottom-right pixel
+                    if (xP < padding && yP < padding)
+                        to.SetPixel(position.X + xP + width + padding, dataOffset + position.Y + yP + height + padding, bottomRight);
+                }
+            }
+        }
+
+        public List<ImageC> PackTextures()
+        {
+            // Increase dynamically the atlas size for fitting all sprites without wasting space.
+            // For now just one atlas, in the future we can have many atlases.
+
+            int padding = 8;
+
+            //Sort textures for packing order...
+            int[] usedTexturesProcessingOrder = new int[_usedTextures.Count];
+            for (int i = 0; i < _usedTextures.Count; ++i)
+                usedTexturesProcessingOrder[i] = i;
+            Array.Sort(usedTexturesProcessingOrder, new TextureComparer { _usedTextures = _usedTextures });
+
+            bool done;
+            int atlasWidth = 256;
+            int atlasHeight = 256;
+
+            do
+            {
+                RectPacker texPacker = new RectPackerTree(new VectorInt2(atlasWidth, atlasHeight));
+
+                done = true;
+
+                foreach (int UsedTextureIndex in usedTexturesProcessingOrder)
```

Comment:
Why `UsedTextureIndex` is capitalized?

Reply:
Copy and paste from somewhere. Fixed :-)